user-manager/src/test/java/com/qpay/usermanager/controller/CustomerControllerIntegrationTest.java

```diff
@@ -0,0 +1,153 @@
+package com.qpay.usermanager.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.qpay.usermanager.model.dto.customer.CustomerModification;
+import com.qpay.usermanager.model.entity.customer.CustomerEntity;
+import com.qpay.usermanager.service.exception.CustomerNotFoundException;
+import com.qpay.usermanager.service.impl.CustomerServiceImpl;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+
+
+@WebMvcTest(controllers = CustomerController.class)
+class CustomerControllerIntegrationTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @MockBean
+    private CustomerServiceImpl customerService;
+
+    private final CustomerEntity CUSTOMER_ENTITY = CustomerEntity
+            .builder()
+            .name("Roman")
+            .email("admin@gmail.com")
+            .password("password")
+            .walletId(1L)
+            .build();
+
+    private final CustomerModification CUSTOMER_MODIFICATION = CustomerModification
+            .builder()
+            .name("Roman")
+            .email("admin@gmail.com")
+            .password("password")
+            .walletId(1L)
+            .build();
+
+    @Test
+    void should_returnCustomer() throws Exception {
+        // given
+        var email = "admin@gmail.com";
+        given(customerService.getCustomerByEmail(email)).willReturn(CUSTOMER_ENTITY);
+
+        // when
+        var responseBody = mockMvc
+                .perform(MockMvcRequestBuilders.get("/api/v1/customers/{email}", email))
+                .andReturn()
+                .getResponse()
+                .getContentAsString();
+
+        // then
+        var expectedResponseBody = objectMapper.writeValueAsString(CUSTOMER_ENTITY);
+        assertThat(responseBody).isEqualTo(expectedResponseBody);
+    }
+
+    @Test
+    void should_returnException_when_CustomerDoesNotExist() throws Exception {
+        // given
+        var email = "qwerty@gmail.com";
+        given(customerService.getCustomerByEmail(email)).willThrow(new CustomerNotFoundException());
+
+        // when
+        var status = mockMvc
+                .perform(MockMvcRequestBuilders.get("/api/v1/customers/{email}", email))
+                .andReturn()
+                .getResponse()
+                .getStatus();
+
+        // then
+        assertThat(status).isEqualTo(404);
```

Comment:
instead of 404 you can use static variables from from spring  or any other libraries. e.g httpStatusNOT_FOUND